federatedscope/db/algorithm/hdtree.py

```diff
@@ -0,0 +1,185 @@
+import federatedscope.db.data.data as data
+import federatedscope.db.model.data_pb2 as datapb
+from federatedscope.db.algorithm.ldp import LDPOLH
+
+import itertools
+import random
+
+class RangeTree(object):
```

Comment:
Please add necessary comments to introduce the classes or methods, and we will use it to generate APIs document automatically. 
```
\```
This is an implementation of range tree refers to [xxx, et al](https://xxxx).

Args: 
    arg1 (int): xxxxx

Returns:
    xxxxx
\```
```